big_scape/utility/filters.py

```diff
@@ -36,3 +39,110 @@ def domain_includelist_filter(run: dict, all_bgc_records: list[bs_gbk.BGCRecord]
                 domainlist_bgc_records.append(record)
 
         return domainlist_bgc_records
+
+
+def generic_category_class_filter(
+    class_categ: set[str], include: set[str], exclude: set[str]
+) -> bool:
+    """Generic function for both class and category based filtering of records
```

Comment:
rewrite so its clear that the input class_categ set is from 1 record alone. and that the main point of this function is to return true if record is to be kept, or false if record is to be excluded